examples/vl3_basic/vl3-nse/cmd/service_registry.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/tiswanso/examples/api/serviceregistry"
+
+	"google.golang.org/grpc"
+)
+
+const (
+	POD_NAME = "podName"
+	SERVICE_NAME = "service"
+	PORT = "port"
+	CLUSTER_NAME = "clusterName"
+)
+
+type validationErrors []error
+
+func NewServiceRegistry(addr string) (ServiceRegistry, ServiceRegistryClient, error) {
+	conn, err := grpc.Dial(addr, grpc.WithInsecure())
```

Comment:
since this is being done synchronously every network-service client (NSC) connection and inside the request `context`, we should take a context and use a dial method that takes context.  Have you tested with a setup that the server isn't available and does this timeout ok?

I think it'd be better to use `tools.DialContextTCP()` or possibly `tools.DialTCPInsecure()`.  I notice the config_ipam.go has the same pattern as this one.  Possibly we can create an issue to follow-up on this connection setup stuff.